packages/mc-scripts/src/commands/config-sync.ts

```diff
@@ -110,6 +115,8 @@ const configSync = async () => {
       data,
     });
 
+    if (!createdCustomApplication) return;
```

Comment:
Not sure about this line. I don't know when the `createCustomApplication` can return a null value and have not raised an error.
Even in that scenario, we would just exit the process and not tell the use what happened.

I guess we would need to know when this use case can happen and, if it really exists, add an explanatory message before exiting the process.

Reply:
The graphql client throws an error, so we don't need to explicitly handle this case.
Not sure if this check is really necessary. Maybe TS complains about something? Could you double check?

Reply:
So, that line was included in [this commit](https://github.com/commercetools/merchant-center-application-kit/commit/2edba6e21e25d93d3370f881e52518b8b81bb779) due to some Typescript checking;

`createdCustomApplication` is the result of calling `createCustomApplication` function which returns an optional object.

This is the function implementation:
```

const createCustomApplication = async ({
  mcApiUrl,
  token,
  organizationId,
  data,
}: TCreateCustomApplicationOptions) => {
  const variables = {
    organizationId,
    data,
  };

  try {
    const createdCustomAppData = await graphQLClient(mcApiUrl, token).request<
      TCreateCustomApplicationFromCliMutation,
      TCreateCustomApplicationFromCliMutationVariables
    >(CreateCustomApplicationFromCli, variables);
    return createdCustomAppData.createCustomApplication;
  } catch (error) {
    if (error instanceof Error) {
      throw new Error(error.message);
    }
    throw error;
  }
};
```

And this is the definition for the `TCreateCustomApplicationFromCliMutation` type:
```
export type TCreateCustomApplicationFromCliMutation = {
  __typename?: 'Mutation';
  createCustomApplication?: {
    __typename?: 'RestrictedCustomApplicationForOrganization';
    id: string;
  } | null;
};
```

So, the return statement `createdCustomAppData.createCustomApplication` can be technically _null_.

If we are sure that sending the create mutation will either work or raise an error, perhaps we can change the type definition to state `createCustomApplication` field is not optional.

This is the mutation sent:
```
mutation CreateCustomApplicationFromCli(
  $organizationId: String!
  $data: CustomApplicationDraftDataInput!
) {
  createCustomApplication(organizationId: $organizationId, data: $data) {
    id
  }
}
```

Reply:
@Rhotimee What's your take about updating the `TCreateCustomApplicationFromCliMutation` so the `createCustomApplication` is not optional?

Reply:
These types are generated so we can't change them. I'll check again if we can make the return type non optional.

Reply:
> @Rhotimee What's your take about updating the `TCreateCustomApplicationFromCliMutation` so the `createCustomApplication` is not optional?

Sorry, I missed the tagged comment. 

The line is not necessary, IMO. It was just added because of the type check error. 
I don't think it matters what we return there because an error will be thrown anyway if the custom application is not created.